### PR TITLE
[FE-2101] Switch rate limiting library

### PR DESCRIPTION
**Pull request by henryfauna**
### Notes
[Jira Ticket](https://faunadb.atlassian.net/browse/FE-2101)

Switch rate limiting library from [limiter](https://www.npmjs.com/package/limiter) to [rate-limiter-flexible](https://www.npmjs.com/package/rate-limiter-flexible).

### CODE REVIEW

File: src/lib/fauna-import-writer.js [hunk 1]
```diff
@@ -37,10 +37,22 @@ function getFaunaImportWriter(
...
+
   const waitForRateLimitTokens = (tokens, rateLimiter) => {
-    while (!rateLimiter.tryRemoveTokens(tokens)) {
+    const tryToConsumeTokens = () => {
+      return rateLimiter
+        .consume('', tokens)
+        .then(() => true)
+        .catch(() => false)
+    }
+    while (!tryToConsumeTokens()) {
```
Comment (cleve-fauna) [inline src/lib/fauna-import-writer.js:55]: `const tryToConsumeTokens` is a promise.

Evaluating its truthiness will always evaluate to true.

You need to use async / await to acheive your goal here of waiting until there's enough tokens.

Comment (cleve-fauna) [reply src/lib/fauna-import-writer.js:55]: A test proving the rate limiter's behavior would have caught this. Its probably worth adding one.

Comment (cleve-fauna) [reply src/lib/fauna-import-writer.js:55]: a trick i've used to test rate limiters before is something like:

```
start timer
set rate limit to like 1 / s
issue ~3 requests
confirm timer >= 3
```

usually pretty easy to code up.

**Review by mwilde345 — APPROVED** [review]

**Review by cleve-fauna — APPROVED** [review]
This is a good start.

I think we should go ahead and do another commit / PR where we consume the remainder of the work in the case the `Math.min(bytesPerSecondLimit, dataSize)` uses the bytesPersSecondLimit (more specifically dataSize > bytePerSecondLmit).

Otherwise we leave ourselves open to letting something we shouldn't through.

**Pull request merged**